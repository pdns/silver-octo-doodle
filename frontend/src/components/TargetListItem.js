import React from 'react';
import PropTypes from 'prop-types';
import {
  withStyles,
  ListItem,
  ListItemIcon,
  ListItemText,
  Typography,
} from '@material-ui/core';
import TargetIcon from 'mdi-material-ui/Target';

const styles = {
  truncateOverflow: {
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
  },
};

function TargetListItem(props) {
  const { target, classes } = props;

  return (
    <ListItem button>
      <ListItemIcon>
        <TargetIcon />
      </ListItemIcon>
      <ListItemText inset className={classes.truncateOverflow}>
        <Typography className={classes.truncateOverflow}>
          {target.name}
        </Typography>
      </ListItemText>
    </ListItem>
  );
}

TargetListItem.propTypes = {
  target: PropTypes.object,
  classes: PropTypes.object,
};

export default withStyles(styles)(TargetListItem);
