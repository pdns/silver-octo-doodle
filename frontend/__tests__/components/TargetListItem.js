/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import TargetListItem from '../../src/components/TargetListItem';

jest.mock('react-relay', () => ({
  createFragmentContainer: x => x,
}));

jest.mock('@material-ui/core/ListItem', () => 'ListItem');
jest.mock('@material-ui/core/ListItemIcon', () => 'ListItemIcon');
jest.mock('@material-ui/core/ListItemText', () => 'ListItemText');
jest.mock('mdi-material-ui/Target', () => 'TargetIcon');

describe('TargetListItem', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<TargetListItem target={{ name: 'target name' }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
