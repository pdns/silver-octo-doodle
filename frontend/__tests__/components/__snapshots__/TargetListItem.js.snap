// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetListItem should render correctly 1`] = `
<ListItem
  button={true}
>
  <ListItemIcon>
    <TargetIcon />
  </ListItemIcon>
  <ListItemText
    inset={true}
    primary="target name"
  />
</ListItem>
`;
